src/main/java/com/google/api/codegen/transformer/csharp/CSharpStandaloneSampleTransformer.java

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.csharp;
+
+import com.google.api.codegen.config.GapicInterfaceContext;
+import com.google.api.codegen.config.GapicProductConfig;
+import com.google.api.codegen.config.InterfaceContext;
+import com.google.api.codegen.config.ProtoApiModel;
+import com.google.api.codegen.config.SampleSpec;
+import com.google.api.codegen.transformer.FileHeaderTransformer;
+import com.google.api.codegen.transformer.ModelToViewTransformer;
+import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.SampleFileRegistry;
+import com.google.api.codegen.transformer.SampleTransformer;
+import com.google.api.codegen.transformer.StandardImportSectionTransformer;
+import com.google.api.codegen.transformer.StaticLangApiMethodTransformer;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.util.Name;
+import com.google.api.codegen.util.csharp.CSharpAliasMode;
+import com.google.api.codegen.viewmodel.MethodSampleView;
+import com.google.api.codegen.viewmodel.StaticLangApiMethodView;
+import com.google.api.codegen.viewmodel.StaticLangFileView;
+import com.google.api.codegen.viewmodel.StaticLangSampleClassView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+
+/** A transformer that generates C# standalone samples. */
+public class CSharpStandaloneSampleTransformer implements ModelToViewTransformer<ProtoApiModel> {
+
+  private static final String STANDALONE_SAMPLE_TEMPLATE_FILENAME = "csharp/standalone_sample.snip";
+  private static final String CSHARP_SAMPLE_PACKAGE_NAME = "Snippets";
```

Comment:
Why `"Snippets"`? Maybe it's more consistent to call it "Samples" or "StandaloneSamples". I think (not sure, @chrisdunelm can confirm) that "snippets" is the term C# is using for their current version of in-code samples.

Reply:
The name is from @beccasaurus 's [canonical sample](https://github.com/SurferJeffAtGoogle/csharp-docs-samples/blob/aa405e94f5e19da8d108336bffe5ddb2ac0ee0b6/dlp/api/Snippets/DeidentifyMasking.cs). But I do see the confusion with incode samples here :/

Reply:
Let's not block this PR, but let's double check whether we  can use "samples" for consistency

Reply:
Package name... hmm. Not passionate at all, developers don't see it.

How about: `Language.v1beta2.Samples.csproj`?

`Talent.v4beta1.Samples.csproj`
`Speech.v1.Samples.csproj`

Reply:
@beccasaurus I'm gonna change it to `Samples` for now. Let's reconsider the structure when working on canonical samples?